.github/workflows/CI.yml

```
@@ -14,5 +14,13 @@ jobs:
     - uses: actions/checkout@v4
     - name: Build
       run: cargo build --verbose
+    - name: Add fmt
+      run: rustup component add rustfmt
+    - name: Add Clippy
+      run: rustup component add clippy
+    - name: Format
+      run: cargo fmt --all -- --check
+    - name: Clippy
+      run: cargo clippy -- -D warnings
```

You may re-order this to deal with fuckups a little bit faster.
```suggestion
    - name: Add fmt
      run: rustup component add rustfmt
    - name: Format
      run: cargo fmt --all -- --check
    - name: Add Clippy
      run: rustup component add clippy
    - name: Clippy
      run: cargo clippy -- -D warnings
```